### PR TITLE
ui/custom-metadata: Cypress tests for custom lists and fields

### PR DESCRIPTION
**Pull request by ssangervasi**
Contributes to #2545

### Code Changes

- ui/cy: Clean up taxonomy fixtures
- ui/flags: Enable customFields in test env
- ui/custom-metadata: Use IconButton for list removal
- ui/custom-metadata: Cypress tests for creating custom lists
- ui/inputs: Allow CustomSelect to take form label props
- ui/custom-metadata: Use CustomSelect - a more standard form field
- ui/custom-metadata: Cypress tests for creating custom fields

### Steps to Confirm

* [ ] Use cypress :)
* [ ] Use the Plus server with https://github.com/ethyca/fidesplus/pull/581 merged in.
    * The selectors are the main thing that had to be changed to use the right components

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`

### Description Of Changes

This mostly just gets the test coverage started so that while I'm hacking into the rest of the forms I can prevent regressions.

I did swap out which Select component is used, because we're trying to normalize our inputs and this one has a lot of conveniences.
custom-field-select

### CODE REVIEW

File: clients/admin-ui/cypress/fixtures/taxonomy/custom-metadata/allow-list/list.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+[
+  {
+    "name": "Pokemon",
+    "description": "There's like a thousand of these now",
```
Comment (LKCSmith) [inline clients/admin-ui/cypress/fixtures/taxonomy/custom-metadata/allow-list/list.json:4]: 😄

File: clients/admin-ui/src/features/common/custom-fields/CreateCustomFields.tsx [hunk 2]
```diff
@@ -1,29 +1,17 @@
...
   Field,
   FieldInputProps,
-  FieldMetaProps,
   Form,
   Formik,
   FormikHelpers,
   FormikProps,
 } from "formik";
-import { ChangeEvent, useCallback, useImperativeHandle, useRef } from "react";
+import { satisfier } from "narrow-minded";
```
Comment (LKCSmith) [inline clients/admin-ui/src/features/common/custom-fields/CreateCustomFields.tsx:10]: 👀 This is yours, right? Nice!

File: clients/admin-ui/src/features/common/custom-fields/form/CustomInput.tsx [hunk 3]
```diff
@@ -17,6 +18,21 @@ import { FieldHookConfig, useField } from "formik";
 
 import { InputType } from "./types";
 
+/**
+ * This isn't the cleanest way to style these forms, but it's a quick hack. Ideally our form inputs
+ * (like CustomSelect) would all use the same styles, or at least have a limited set of variants:
+ * https://chakra-ui.com/docs/styled-system/component-style#styling-multipart-components
+ *
+ * Unfortunately the designs for these forms call for styles that are inconsistent with the other
```
Comment (LKCSmith) [inline clients/admin-ui/src/features/common/custom-fields/form/CustomInput.tsx:26]: Has this concern been brought to Design?

**Review by LKCSmith — APPROVED** [review]
👏 🎉

File: clients/admin-ui/src/features/common/custom-fields/form/CustomInput.tsx (hunk 3 above)
Comment (ssangervasi) [reply clients/admin-ui/src/features/common/custom-fields/form/CustomInput.tsx:26]: Here's some screenshots of the difference.

How most of our labels look:
Screen Shot 2023-02-10 at 10 36 16

How the labels in the modal look (smaller, bolder, grayer):
Screen Shot 2023-02-10 at 10 36 24

File: clients/admin-ui/src/features/common/custom-fields/CreateCustomFields.tsx [hunk 4]
```diff
@@ -87,43 +67,45 @@ const CreateCustomFields = forwardRef(
...
-              initialValues.allow_list_id = value.id;
-            }
-          }
-        });
-      }
-      list.set("Select...", { value: "" });
-      return list;
-    }, [data]);
-
-    const list = loadList();
```
Comment (ssangervasi) [inline clients/admin-ui/src/features/common/custom-fields/CreateCustomFields.tsx:106]: [Post-review]: This wasn't memoizing. `useCallback` returns a closure that only changes when the dependency list changes. However, the _return value_ of that function isn't memoized, just the function itself. So this code was still calling the function on every render, doing all of that work, and returning a new list, even when the dependencies hadn't changed.

File: clients/admin-ui/src/features/common/custom-fields/CreateCustomFields.tsx [hunk 5]
```diff
@@ -87,43 +67,45 @@ const CreateCustomFields = forwardRef(
...
     };
 
-    const loadList = useCallback((): Map<string, ItemOption> => {
-      const list = new Map<string, ItemOption>();
-      if (data?.length) {
-        data.forEach((value, index) => {
-          if (value?.id && value?.name) {
-            list.set(value.name, { value: value.id });
-            if (index === 0) {
-              initialValues.allow_list_id = value.id;
```
Comment (ssangervasi) [inline clients/admin-ui/src/features/common/custom-fields/CreateCustomFields.tsx:97]: [Post-review]: Mutating object like this is _very_ error prone. Not only was this a dirty way of populating the object on first render, but this was also modifying a **module scoped constant**. So this value could stick around even when the component was rendered for a completely different field.

File: clients/admin-ui/src/features/common/custom-fields/CreateCustomFields.tsx [hunk 6]
```diff
@@ -49,30 +37,22 @@ const initialValues: CustomFieldDefinition = {
   resource_type: ResourceTypes.DATA_CATEGORY,
 };
 
-type FormValues = typeof initialValues;
```
Comment (ssangervasi) [inline clients/admin-ui/src/features/common/custom-fields/CreateCustomFields.tsx:52]: [Post-review]: Redundant alias for CustomFieldDefinition that was already being used.

File: clients/admin-ui/src/features/common/custom-fields/CreateCustomFields.tsx [hunk 7]
```diff
@@ -158,109 +140,25 @@ const CreateCustomFields = forwardRef(
...
-                          }}
-                          onChange={(value: string | undefined) => {
-                            props.setFieldValue("resource_type", value);
-                          }}
-                          selectedValue={props.values.resource_type}
-                        />
-                      </VStack>
-                    </FormControl>
-                  )}
-                </Field>
```
Comment (ssangervasi) [inline clients/admin-ui/src/features/common/custom-fields/CreateCustomFields.tsx:263]: [Post-review]: The `SelectDropdown` was an old component. `CustomSelect` has more functionality.

File: clients/admin-ui/src/features/common/custom-fields/form/CustomInput.tsx [hunk 8]
```diff
@@ -17,6 +18,21 @@ import { FieldHookConfig, useField } from "formik";
...
+ * forms in the app.
+ */
+export const CUSTOM_LABEL_STYLES: FormLabelProps = {
+  color: "gray.600",
+  fontSize: "14px",
+  fontWeight: "semibold",
+  minWidth: "150px",
+};
```
Comment (ssangervasi) [inline clients/admin-ui/src/features/common/custom-fields/form/CustomInput.tsx:34]: [Post-review]: The whole component below should go away in favor of:
- Using input fields from `~/features/common/form/inputs`
- Adding the ability to override label styles ([for now](https://ethyca.slack.com/archives/C043URV1KLL/p1676061654400279?thread_ts=1676054836.626409&cid=C043URV1KLL))
- Extracting the useful parts of the number input to that module so we could use it elsewhere if needed.

**Pull request merged**